src/combat/combat.js

```diff
@@ -11,13 +12,82 @@ export default class YearZeroCombat extends Combat {
    */
   async rollInitiative(ids, options) {
     ids = typeof ids === 'string' ? [ids] : ids;
-    // TODO
+    const initiativeDeck = getInitiativeDeck(true);
+    const { messageOptions } = options;
+    const messages = [];
+
+    if (ids.length > initiativeDeck.availableCards.length) {
+      const message = game.i18n.format('YZE.Combat.Initiative.NotEnoughCards', {
+        count: ids.length,
+        available: initiativeDeck.availableCards.length,
+      });
+      ui.notifications.warn(message);
+      return this;
+    }
+
+    for (const id of ids) {
+      const combatant = this.combatants.get(id, true);
+      if (combatant.isDefeated) continue;
+      const drawSize = 1;
+      // TODO add hooks for more than one card to be drawn
+      // let card;
+      const cards = await initiativeDeck.draw(drawSize);
+      // TODO add logic for how we handle multiple cards
+      const card = cards[0];
+      const flags = {
+        cardValue: card.data.value,
+        // cardString: card.data.name, // here if we need it!
+      };
+      const initiative = card?.data.value;
+      combatant.data.update({ initiative: initiative, 'flags.yze-combat': flags });
+
+      const template = `
+      <section class="initiative-card">
+      <div class="initiative-card-container">
+      <img class="initiative-card-image" src="${card?.face?.img}" />
+      </div>
+      <h4 class="result-text result-text-card">${card?.name}</h4>
+      </section>`;
+
+      const messageData = mergeObject(
+        {
+          speaker: {
+            scene: game.scenes?.active?.id,
+            actor: combatant.actor ? combatant.actor.id : null,
+            alias: `${combatant.token.name} ${game.i18n.localize('YZE.Combat.Initiative.Draw')}`,
```

Comment:
```suggestion
            alias: game.i18n.format('YZE.Combat.Initiative.Draw', { name: combatant.token.name }),
```
Due to incompatibilities between languages, it's better to put the token name in the translation variable using the `.format` method.

I'll also add the modification for the YML translation files.